gnosis/cowswap
# Add envs ipfshash

# Summary

This PR does mainly two things:
* Uses a different AppData code per environemnt
* Makes further differentiation introducing new environments: `Local`, `PR`

![image](https://user-images.githubusercontent.com/2352112/130063090-956c9e69-aea7-4933-b6ef-684addfc51c3.png)

![image](https://user-images.githubusercontent.com/2352112/130064086-608fef98-bd05-4fb0-9d80-fc2180058762.png)


Note that it also adds a new metadata `environment` in the App Data JSON file:
```json
{
  "version": "0.1.0",
  "appCode": "CowSwap",
  "metadata": {
    "environment": "barn"
  }
}
```

Additionally, it renames `prestaging` environment to `barn`, just to be more clear.

The environment is exposed also as a global var (window.environment), to make it easier to read. 
This PR also affects to sentry, now we have more environments, so we have further details where the errors come from.

![image](https://user-images.githubusercontent.com/2352112/130064207-dc68ef85-5e1d-4c10-b3c6-3b34ab1455e4.png)


# To Test

1. Open the app link, and check in the console that the environment is `pr`. 
2. Sign an order, and verify the `appData` you sign matches the `pr` one.

Test also locally, and check in the console that the environment is `local`. 
1. Run app locally
2. Sign an order, and verify the `appData` you sign matches the `local` one.

# Background
- We want to segment the volume also by environment if needed.
- Also nice to differentiate in sentry, so we know if the erros are from dev, pr or local
